Zend/zend_ast.c

```diff
@@ -1109,9 +1109,11 @@ static ZEND_COLD void zend_ast_export_encaps_list(smart_str *str, char quote, ze
 		           ast->child[0]->kind == ZEND_AST_ZVAL &&
 		           (i + 1 == list->children ||
 		            list->child[i + 1]->kind != ZEND_AST_ZVAL ||
+					(*Z_STRVAL_P(
+					    zend_ast_get_zval(list->child[i + 1])) != '[' &&
 		            !zend_ast_valid_var_char(
 		                *Z_STRVAL_P(
-		                    zend_ast_get_zval(list->child[i + 1]))))) {
+		                    zend_ast_get_zval(list->child[i + 1])))))) {
```

Comment:
This code is becoming quite hard to read. Can you maybe extract a function `zend_ast_var_needs_braces()` that performs these checks?